utility/types/message.go

```diff
@@ -267,3 +278,12 @@ func ValidateStaker(msg MessageStaker) types.Error {
 	}
 	return ValidateServiceUrl(msg.GetActorType(), msg.GetServiceUrl())
 }
+
+func getSignBytes(msg Message) []byte {
```

Comment:
We must sort them if json.

However, it appears in your implementation you are using protobuf.

I encourage us to consider protobuf for this task - no sorting is required if this is the case.

Reply:
Basically we want deterministic serialization of the payloads and a different ordering of the fields in a JSON can result in different strings/byte arrays.
In the current implementation I am relying on the protobuf generated structs and I am calling the Marshal method, however I just realized that that's not deterministic by default.
In fact in https://github.com/deblasis/pocket/blob/b80a57c00310fe70399350879cd1fd848d0d8b07/shared/types/codec.go#L20 we call `proto.Marshal(message)` but really we should call `proto.MarshalOptions{Deterministic: true}.Marshal(message)` however, as documented here https://pkg.go.dev/google.golang.org/protobuf/proto 
![image](https://user-images.githubusercontent.com/29378614/188329146-0a29b489-df54-4abd-9016-81e6aea9812a.png)
Using `Deterministic:true` is probably not a strong guarantee that in the future this won't change. However I noticed that for example  Cosmos approached the problem in Pulsar https://github.com/cosmos/cosmos-proto/search?q=deterministic using  `proto.MarshalOptions{Deterministic: true}.Marshal(message)`
What do you guys think we should do? It seems important and part of a bigger scope, new issue?

Reply:
Tracked as #201